src/components/EditNotePage.js

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import {connect} from "react-redux";
+import {bindActionCreators} from "redux";
+import * as actions from "../actions";
+import {Button} from "react-bootstrap";
+
+class EditNotePage extends React.Component {
+
+  state = {
+    note: {
+      title: '',
+      message: ''
+      
+    }
+  };
+
+  componentDidMount() {
+    const noteId = this.props.match.params.id;
+
+    const note = this.props.notes.filter(e => e.id === Number(noteId))[0];
+    this.setState({note});
+  }
+
+  setTitle = (event) => {
+    const note = this.state.note;
+    note.title = event.target.value;
+    this.setState({note});
+  };
+
+  setMessage = (event) => {
+    const note = this.state.note;
+    note.message = event.target.value;
+    this.setState({note});
```

Comment:
I would recommend looking into using functional set state when you're rendering the new state using part of the old state.  [Good article](https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b)